examples/lwip_ethernet_tap/main.c

```diff
@@ -0,0 +1,438 @@
+  // receive the frame, but the kernel reasserts the flag, which simply will
+  // have the app swap the buffer again. If we were to do this after swapping
+  // the buffer, we might deassert the flag despite there being a pending
+  // frame waiting for reception:
+  state->rx_frame_received = false;
+
+  // Swap buffers and check if there is any new payload:
+  uint8_t* current_ptr;
+  uint32_t remaining_len;
+  returncode_t res =
+    streaming_process_slice_get_and_swap(&state->rx_streaming_state, &current_ptr, &remaining_len, NULL);
+  if (res != RETURNCODE_SUCCESS) {
+    printf("-> failed swapping frame receive buffer: %d\r\n", res);
+    return;
+  }
+
+  // Process frames one by one, incrementing `current_ptr` and decrementing
+  // `remaining_len`:
+  while (remaining_len >= 12) {
+    int iface_input_res;
+    uint16_t flags, frame_len;
+    uint64_t receive_ts;
+
+    memcpy(&flags, current_ptr, sizeof(uint16_t));
+    current_ptr   += sizeof(uint16_t);
+    remaining_len -= sizeof(uint16_t);
+
+    memcpy(&frame_len, current_ptr, sizeof(uint16_t));
+    current_ptr   += sizeof(uint16_t);
+    remaining_len -= sizeof(uint16_t);
+
+    memcpy(&receive_ts, current_ptr, sizeof(uint64_t));
+    current_ptr   += sizeof(uint64_t);
+    remaining_len -= sizeof(uint64_t);
+
+    if (frame_len > remaining_len) {
+      printf("-> frame_len larger than remaining_len, illegal: %u vs. %lu\r\n",
+             frame_len, remaining_len);
+      return;
+    }
+
+    LINK_STATS_INC(link.recv);
+
+    // Allocate a frame buffer from the global LWIP pool to copy the
+    // received frame into.
+    struct pbuf* received_frame = pbuf_alloc(PBUF_RAW, frame_len, PBUF_POOL);
+    if (received_frame == NULL) {
+      // We failed to allocate sufficient space in the pool
+      printf("-> input_frames failed to allocate pbuf for %d bytes\r\n",
+             frame_len);
+      current_ptr   += frame_len;
+      remaining_len -= frame_len;
+      continue;
+    }
+
+    // Copy the frame into the frame buffer
+    pbuf_take(received_frame, current_ptr, frame_len);
+    current_ptr   += frame_len;
+    remaining_len -= frame_len;
+
+    iface_input_res = iface->input(received_frame, iface);
+    if (iface_input_res != ERR_OK) {
+      printf("-> iface.input failed, res = %d\r\n", iface_input_res);
+      pbuf_free(received_frame);
+    }
+  }
+
+  // The above loop should always process the entire payload:
+  if (remaining_len != 0) {
+    printf("-> input_frames has trailing bytes after consuming frames in "
+           "streaming process slice (%lu bytes)\r\n",
+           remaining_len);
+  }
+}
+
+static void timeouts_alarm_callback(__attribute__((unused)) uint32_t now,
+                                    __attribute__((unused)) uint32_t scheduled,
+                                    void*                            opaque) {
+  bool* alarm_fired = opaque;
+  *alarm_fired = true;
+}
+
+int main(void) {
```

Comment:
Cool!

Can you add a README describing what this does and what is expected?

Reply:
It would be good to include an example of "it worked on this board at this Tock release"